src/azure-cli/azure/cli/command_modules/rdbms/_flexible_server_util.py

```diff
@@ -297,11 +297,45 @@ def get_current_time():
     return datetime.utcnow().replace(tzinfo=dt.timezone.utc, microsecond=0).isoformat()
 
 
-def change_str_to_datetime(date_str):
-    for fmt in ("%Y-%m-%dT%H:%M:%S+00:00", "%Y-%m-%dT%H:%M:%S.%f+00:00"):
-        try:
-            return datetime.strptime(date_str, fmt)
-        except ValueError:
-            pass
+def get_id_components(rid):
+    parsed_rid = parse_resource_id(rid)
+    subscription = parsed_rid['subscription']
+    resource_group = parsed_rid['resource_group']
+    vnet_name = parsed_rid['name']
+    subnet_name = parsed_rid['child_name_1'] if 'child_name_1' in parsed_rid else None
 
-    raise ValidationError("The format of restore time should be %Y-%m-%dT%H:%M:%S+00:00")
+    return subscription, resource_group, vnet_name, subnet_name
+
+
+def check_existence(resource_client, value, resource_group, provider_namespace, resource_type,
+                    parent_name=None, parent_type=None):
+
+    parent_path = ''
+    if parent_name and parent_type:
+        parent_path = '{}/{}'.format(parent_type, parent_name)
+
+    api_version = _resolve_api_version(resource_client, provider_namespace, resource_type, parent_path)
+
+    try:
+        resource_client.resources.get(resource_group, provider_namespace, parent_path, resource_type, value, api_version)
+    except:  # pylint: disable=bare-except
+        return False
+    return True
+
+
+def _resolve_api_version(client, provider_namespace, resource_type, parent_path):
+    provider = client.providers.get(provider_namespace)
+
+    # If available, we will use parent resource's api-version
+    resource_type_str = (parent_path.split('/')[0] if parent_path else resource_type)
+
+    rt = [t for t in provider.resource_types  # pylint: disable=no-member
+          if t.resource_type.lower() == resource_type_str.lower()]
+    if not rt:
+        raise CLIError('Resource type {} not found.'.format(resource_type_str))
```

Comment:
use error type in azclierror.py https://github.com/Azure/azure-cli/blob/3bff786b9e337aa35bb4cc64a09e5600a35663f8/src/azure-cli-core/azure/cli/core/azclierror.py#L136

Reply:
Modified this.. thanks